File: src/main/java/com/maarek/kosar/Scope.java
package com.maarek.kosar;

import com.maarek.kosar.scope.VisibleScope;

/**
 * Created by jlyman on 10/12/15.
 */
public class Scope extends VisibleScope {
    public static class C4 extends NestedStaticClass {}

    public Scope() {
        new NestedStaticClass();
    }
}

File: src/main/kotlin/com/maarek/kosar/Scope.kt
package com.maarek.kosar

import com.maarek.kosar.scope.HiddenScope
import com.maarek.kosar.scope.VisibleScope

/**
 * Created by jlyman on 10/12/15.
 */

class KotlinScope() : VisibleScope() {
    class C1() : HiddenScope.NestedStaticClass()

    init {
        val v = HiddenScope.NestedStaticClass()
    }
}
